File: app/models/view/system/Menu.go
package v_system

type AddMenuReq struct {
	Name     string `json:"name" valid:"required~菜单名称不能为空"`
	Icon     string `json:"icon" valid:"stringlength(3|18)~图标名称长度为3-12位"`
	RouteID  int    `json:"route_id"`
	ParentID int    `json:"parent_id"`
	Sort     int    `json:"sort"`
	Enabled  bool   `json:"enabled"`
}

type UpdateMenuReq struct {
	ID int `json:"id" valid:"int~菜单ID必须为整型数字,gt(0)~菜单ID必须大于0"`
	AddMenuReq
}

File: fe/src/api/system/menu.ts
import request from '@/utils/request';
import { IFrontendRoute } from '@/api/system/frontend-route'

export interface IMenu {
    id?: number;
    name: string;
    icon?: string;
    route_id: number;
    parent_id: number;
    enabled: boolean;
    sort: number;
    children?: IMenu[];
    route?: IFrontendRoute;
}

export const listTreeApi = (): IRespData<Required<IMenu>[]> => request.get("/Menu/ListTree")
export const createApi = (data: IMenu): IRespData<null> => request.post("/Menu/Add", data)
export const deleteApi = (id: number): IRespData<null> => request.delete("/Menu/Delete", { params: { id } })
export const editApi = (data: IMenu): IRespData<null> => request.post("/Menu/Edit", data)
export const updateStatus = (data: { id: number, enabled: boolean }): IRespData<null> => request.post("/Menu/UpdateStatus", data)
export const getApi = (id: number): IRespData<Required<IMenu>> => request.get("/Menu/Get", { params: { id } })


// 递归查询指定菜单列表中是否存在路由路径等于指定路由路径的菜单
export function existsByRoutePath(menus: IMenu[] | undefined, path: string): boolean {
    if (menus === undefined) return false;
	for (let menuItem of menus) {
		if (menuItem.route?.route_path === path) {
			return true
		}
		if (menuItem.children && existsByRoutePath(menuItem.children, path)) {
			return true
		}
	}
	return false
}